### PR TITLE
JETUT-21 Fixed Github Actions

### PR DESCRIPTION
**Pull request by kito99**

### CODE REVIEW

**Comment by BalusC** [comment]
The change at its own looks good but do these Github Action files really belong in this repo? It seems really to me that these belong to the https://github.com/virtua-tech/jakartaee-tutorial repo. I have doublechecked the pom.xml and README of both repos and I really don't think these Github Action files belong in this repo.

**Comment by BalusC** [comment]
If these files really belong here then please remove/change the phrase "release/10" in the deploy file https://github.com/virtua-tech/jakartaee-tutorial-playbook/blob/main/.github/workflows/deploy.yml#L11

**Review by BalusC — APPROVED** [review]

**Review by BalusC — APPROVED** [review]

File: .github/workflows/build-and-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - id: checkout
+        name: Checkout
+        uses: actions/checkout@v3
+      - id: install-jhdk
```
Comment (BalusC) [inline .github/workflows/build-and-deploy.yml]: typo

File: .github/workflows/build-and-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+          cache: maven
+      - id: build
+        name: Build Docs
+        run: mvn -B clean verify
+      - id: create-gh-pages-artifact
+        name: Create Artifact
+        if: github.ref == 'refs/heads/main'
```
Comment (BalusC) [inline .github/workflows/build-and-deploy.yml]: unnecessary condition, this is always true for the currently configured 'on', perhaps you wanted to check github.event == 'push' instead?

File: .github/workflows/build-and-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        uses: actions/upload-pages-artifact@v1
+        with:
+          path: target/generated-docs
+  deploy:
+    name: Deploy to GitHub Pages
+    if: github.ref == 'refs/heads/main'
```
Comment (BalusC) [inline .github/workflows/build-and-deploy.yml]: unnecessary condition, this is always true for the currently configured 'on', perhaps you wanted to check github.event == 'push' instead?

Comment (kito99) [reply .github/workflows/build-and-deploy.yml]: Good catch!

**Pull request merged**